stl/inc/xutility

```diff
@@ -701,6 +697,13 @@ struct projected {
     }
 };
 
+#if _HAS_CXX23
+template <class _It, class _Proj>
+struct _Indirect_value_impl<projected<_It, _Proj>> {
+    using type = invoke_result_t<_Proj&, _Indirect_value_t<_It>>;
+};
+#endif // _HAS_CXX23
+

```

Comment:
To be clear: `_Indirect_value_t<T>` in C++20 mode is same as `iter_value_t<T>&`, and only in C++23 mode we add this extra specialization.